pkg/test/echo/client/client.go

```diff
@@ -52,7 +52,13 @@ func New(address string, tlsSettings *common.TLSSettings) (*Instance, error) {
 			return nil, err
 		}
 
-		certPool := x509.NewCertPool()
+		var certPool *x509.CertPool
+		if tlsSettings.RootCert == "" {
+			certPool, _ = x509.SystemCertPool()
+		} else {
+			certPool.AppendCertsFromPEM([]byte(tlsSettings.RootCert))
```

Comment:
the `certPool` is nil if RootCert != "", you should call NewCertPool() to create it first.